model/sparql/GenericSparql.php

```diff
@@ -1429,6 +1450,55 @@ public function queryLabel($uri, $lang) {
         }
     }
 
+    /**
+     * Query for skos:prefLabels, skos:altLabels and skos:hiddenLabels of a resource.
+     * @param string $uri
+     * @param string $lang
+     * @return array array of prefLabels, altLabels and hiddenLabels - or null if resource doesn't exist
+     */
+    public function queryOtherLabels($uri, $lang) {
+        $query = $this->generateOtherLabelQuery($uri, $lang);
+        $result = $this->query($query);
+        $ret = array();
+
+        $prefLabels = array();
+        $altLabels = array();
+        $hiddenLabels = array();
+
+        foreach ($result as $row) {
```

Comment:
How about replacing the contents of the `foreach` loop with something like:

```php
$lname = $row->prop->localName();
if (!in_array($lname, $ret) {
    $ret[$lname] = array();
}
$ret[$lname][] = $row->val;
```

Then you don't need the arrays `$prefLabels`, `$altLabels` and `$hiddenLabels` at all, and can also drop the three `if (!empty(...))` clauses below.

Reply:
That is certainly much more elegant, thanks for the suggestion.

Reply:
I did the changes you recommended. It seems much cleaner now.